DemoServer/Controllers/Demos/JavascriptIndexes/JavascriptMapIndex/JavascriptMapIndexController.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System.Threading.Tasks;
+using DemoCommon.Models;
+using DemoServer.Utils.Cache;
+using DemoServer.Utils.Database;
+using DemoServer.Utils.UserId;
+using Microsoft.AspNetCore.Mvc;
+#region Usings
+using System.Linq;
+using System.Collections.Generic;
+using Raven.Client.Documents.Session;
+using Raven.Client.Documents.Indexes;
+#endregion
+
+namespace DemoServer.Controllers.Demos.JavascriptIndexes.JavascriptMapIndex
+{
+    public class JavascriptMapIndexController : DemoCodeController
+    {
+        public JavascriptMapIndexController(UserIdContainer userId, UserStoreCache userStoreCache, MediaStoreCache mediaStoreCache,
+            DatabaseSetup databaseSetup) : base(userId, userStoreCache, mediaStoreCache, databaseSetup)
+        {
+        }
+        
+        #region Demo
+        #region Step_1
+        public class Employees_ByImportantDetailsJS : AbstractJavaScriptIndexCreationTask
+            #endregion
+        {
+            #region Step_2
+            public class IndexEntry
+            {
+                public string FullName { get; set; }
+                public string Country { get; set; }
+                public int    WorkingInCompanySince { get; set; }
+                public int    NumberOfTerritories { get; set; }
+            }
+            #endregion
+            
+            #region Step_3
+            public Employees_ByImportantDetailsJS()
+            {
+                Maps = new HashSet<string>
+                {
+                    @"map('Employees', function (employee) {
```

Comment:
What do you think about a bit more modern syntax where instead of

`function (employee) {`
you go with
`employee => {`

Reply:
as discussed, prefer to leave this with 'function'